src/server/reply.rs

```diff
@@ -1,37 +1,95 @@
 use serde::Serialize;
+use warp::http::header::HeaderValue;
 use warp::http::status::StatusCode;
 use warp::reply::Response;
 use warp::Reply;
 
-use super::TOML_MIME_TYPE;
+use tracing::log::debug;
+
+use super::{JSON_MIME_TYPE, TOML_MIME_TYPE};
 use crate::provider::ProviderError;
 
-// Borrowed and modified from https://docs.rs/warp/0.2.5/src/warp/reply.rs.html#102
-pub fn toml<T>(val: &T) -> Toml
+/// Use an accept header to determine how to serialize content.
+///
+/// This will examine the Accept header, looking for the best match, and then it will
+/// use the appropriate serializer to serialize the data.
+///
+/// The current implementation ignores `q=` annotations, assigning preference based on
+/// the first MIME type to match.
+///
+/// For example, `Accept: text/json, application/toml;q=0.9` will cause encoding to be in JSON.
+/// If no suitable content type is found, this will encode in application/toml, as that
+/// is the behavior described in the spec.
+pub fn serialized_data<T>(val: &T, accept: String) -> SerializedData
 where
     T: Serialize,
 {
-    Toml {
-        inner: toml::to_vec(val).map_err(|e| {
+    let supported = vec![
```

Comment:
Nit: Consider breaking this out into a const slice to avoid string allocation every time